### PR TITLE
Support Copycat blocks copying blast resistance

### PR DESCRIPTION
**Pull request by Endalion**
Added a `BlockHardnessHandler::getHardness` method which takes Create Copycat blocks into consideration, also checking the block/tag map on the copied block.
The parameters differ so I only changed the resistance tool as an example, if this patch is satisfactory then I'll propagate to the rest

### CODE REVIEW

**Comment by rbasamoyai** [comment]
Could be in 0.5.4, we'll see

**Comment by rbasamoyai** [comment]
Although, I think it may be interesting to add a mixin interface to `CopycatBlock` instead of doing a class type check. This would make mod compat easier for other mods and would allow modifying the return value. FramedBlocks comes to mind here.

**Comment by Endalion** [comment]
Hm, I'm not quite sure how to mixin it that way but I'll try it out

**Comment by Endalion** [comment]
wow mixins are powerful 👍 I also changed the rest of the hardness checks to follow the new method

File: common/src/main/java/rbasamoyai/createbigcannons/mixin/CopycatMixin.java [hunk 1]
```diff
@@ -7,14 +7,15 @@
...
+import org.jetbrains.annotations.NotNull;
 import org.spongepowered.asm.mixin.Mixin;
 
 import rbasamoyai.createbigcannons.munitions.config.PropertiesMimicTypeBlock;
 
 @Mixin(CopycatBlock.class)
 public abstract class CopycatMixin implements PropertiesMimicTypeBlock {
 	@Override
-	public Block createBigCannons$getActualBlock(BlockState state, Level level, BlockPos pos) {
+	public @NotNull Block createBigCannons$getActualBlock(BlockState state, Level level, BlockPos pos) {
```
Comment (rbasamoyai) [inline common/src/main/java/rbasamoyai/createbigcannons/mixin/CopycatMixin.java]: Use Javax nonnull

File: common/src/main/java/rbasamoyai/createbigcannons/munitions/config/PropertiesMimicTypeBlock.java [hunk 2]
```diff
@@ -5,6 +5,8 @@
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
 
+import org.jetbrains.annotations.NotNull;
+
 public interface PropertiesMimicTypeBlock {
-	public Block createBigCannons$getActualBlock(BlockState state, Level level, BlockPos pos);
+	@NotNull Block createBigCannons$getActualBlock(BlockState state, Level level, BlockPos pos);
```
Comment (rbasamoyai) [inline common/src/main/java/rbasamoyai/createbigcannons/munitions/config/PropertiesMimicTypeBlock.java]: Ditto

**Review by rbasamoyai — CHANGES_REQUESTED** [review]
Use javax nonnull instead of JetBrains NotNull

**Review by rbasamoyai — APPROVED** [review]

**Pull request merged**